noodles-editor/src/noodles/components/field-components.tsx

```diff
@@ -1134,6 +1136,185 @@ export function DateFieldComponent({
+        case 's': // Second: 10:30:00 or 2024-03-15T10:30:00
+          return temporalType === 'time' ? str.substring(0, 8) : str.substring(0, 19)
+        case 'ms': // Millisecond: full precision
+          return str.substring(0, 23)
+        default:
+          return str
+      }
+    } catch (e) {
+      console.warn('Error formatting temporal value:', e)
+      return ''
+    }
+  }
+
+  // Parse input value back to appropriate Temporal type
+  const parseFromInput = (inputValue: string) => {
+    if (!inputValue) return
+
+    try {
+      // Pad incomplete values based on precision
+      let paddedValue = inputValue
+
+      // Add missing components based on type and precision
+      if (temporalType === 'date' || temporalType === 'datetime' || temporalType === 'zoned-datetime') {
+        // Ensure we have at least YYYY-MM-DD for dates
+        if (precision === 'y') {
+          paddedValue = `${inputValue}-01-01`
+        } else if (precision === 'M') {
+          paddedValue = `${inputValue}-01`
+        }
+
+        // Add time component for datetime types if needed
+        if (temporalType === 'datetime' || temporalType === 'zoned-datetime') {
+          if (precision === 'd') {
+            paddedValue = `${paddedValue}T00:00:00`
+          } else if (precision === 'h') {
+            paddedValue = `${paddedValue}:00:00`
+          } else if (precision === 'm') {
+            paddedValue = `${paddedValue}:00`
+          }
+        }
+      } else if (temporalType === 'time') {
+        // Pad time components
+        if (precision === 'h') {
+          paddedValue = `${inputValue}:00:00`
+        } else if (precision === 'm') {
+          paddedValue = `${inputValue}:00`
+        }
+      }
+
+      // Parse based on type
+      switch (temporalType) {
+        case 'date':
+          field.setValue(Temporal.PlainDate.from(paddedValue))
+          break
+        case 'time':
+          field.setValue(Temporal.PlainTime.from(paddedValue))
+          break
+        case 'datetime':
+          field.setValue(Temporal.PlainDateTime.from(paddedValue))
+          break
+        case 'zoned-datetime':
+          // Try to parse with timezone, or add one
+          try {
+            field.setValue(Temporal.ZonedDateTime.from(paddedValue))
+          } catch {
+            field.setValue(Temporal.PlainDateTime.from(paddedValue).toZonedDateTime(timeZone))
+          }
+          break
+      }
+    } catch (e) {
+      console.warn('Error parsing temporal value:', e)
+    }
+  }
+
+  const onChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    parseFromInput(e.currentTarget.value)
+  }
+
+  // Determine HTML input type based on temporal type
+  const getInputType = (): string => {
+    switch (temporalType) {
+      case 'date':
+        return 'date'
+      case 'time':
+        return 'time'
+      case 'datetime':
+      case 'zoned-datetime':
+        return 'datetime-local'
```

Comment:
I'm not sure this works. We'd need inputs to be two-part: https://developer.mozilla.org/en-US/docs/Web/HTML/Reference/Elements/input/datetime-local#setting_timezones